bindings/sftp/sftp.go

```diff
@@ -0,0 +1,314 @@
+package sftp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+
+	sftpClient "github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/knownhosts"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/components-contrib/metadata"
+	"github.com/dapr/kit/logger"
+	kitmd "github.com/dapr/kit/metadata"
+)
+
+const (
+	metadataRootPath = "rootPath"
+	metadataFileName = "fileName"
+)
+
+// Sftp is a binding for file operations on sftp server.
+type Sftp struct {
+	metadata   *sftpMetadata
+	logger     logger.Logger
+	sftpClient *sftpClient.Client
+}
+
+// sftpMetadata defines the sftp metadata.
+type sftpMetadata struct {
+	RootPath       string `json:"rootPath"`
+	FileName       string `json:"fileName"`
```

Comment:
We probably don't want the fileName in the _component_ metadata, right? Each operation should be on its own file.